### PR TITLE
List unit users as Super Admin

### PR DESCRIPTION
**Pull request by inaod568**
Super admins can also list unit users. Needs to specify a unit. 

**Test with:** https://github.com/ScilifelabDataCentre/dds_cli/pull/397

- [x] Tests passing
- [x] Black formatting
- [ - ] Migrations for any changes to the database schema
- [ ] Rebase/merge the `dev` branch
- [x] Note in the CHANGELOG

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -50,3 +50,4 @@ Please add a _short_ line describing the PR you make, if the PR implements a spe
 - When listing the projects, return whether or not the user has a project key for that particular project ([#1049](https://github.com/ScilifelabDataCentre/dds_web/pull/1049))
 - New endpoint for Unit Personnel and Admins to list the other Unit Personnel / Admins within their project ([#1050](https://github.com/ScilifelabDataCentre/dds_web/pull/1050))
 - Make previous HOTP invalid at password reset ([#1054](https://github.com/ScilifelabDataCentre/dds_web/pull/1054))
+- New endpoint for listing unit users as Super Admin ()
```
Comment (inaod568) [inline CHANGELOG.md:53]:
```suggestion
- New endpoint for listing unit users as Super Admin ([#1059](https://github.com/ScilifelabDataCentre/dds_web/pull/1059))
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -50,3 +50,4 @@ Please add a _short_ line describing the PR you make, if the PR implements a spe
 - When listing the projects, return whether or not the user has a project key for that particular project ([#1049](https://github.com/ScilifelabDataCentre/dds_web/pull/1049))
 - New endpoint for Unit Personnel and Admins to list the other Unit Personnel / Admins within their project ([#1050](https://github.com/ScilifelabDataCentre/dds_web/pull/1050))
 - Make previous HOTP invalid at password reset ([#1054](https://github.com/ScilifelabDataCentre/dds_web/pull/1054))
+- New endpoint for listing unit users as Super Admin ([#1059](https://github.com/ScilifelabDataCentre/dds_web/pull/1059))
```
Comment (inaod568) [inline CHANGELOG.md:53]:
```suggestion
- New endpoint for listing unit users as Super Admin ([#1059](https://github.com/ScilifelabDataCentre/dds_web/pull/1059))

```

File: dds_web/api/user.py [hunk 3]
```diff
@@ -949,6 +949,23 @@ def get(self):
...
             )
 
+        if auth.current_user().role == "Super Admin":
+            json_input = flask.request.json
+            if not json_input:
+                raise ddserr.DDSArgumentError(message="Unit public id missing.")
+
+            unit = json_input.get("unit")
+            if not unit:
+                raise ddserr.DDSArgumentError(message="Unit public id missing.")
```
Comment (talavis) [inline dds_web/api/user.py:959]: Not sure that it's better than the current approach, but at least it's not repeating the same error message:
```suggestion
            if not json_input or not (unit := json_input.get("unit")):
                raise ddserr.DDSArgumentError(message="Unit public id missing.")
```

File: dds_web/api/user.py [hunk 4]
```diff
@@ -935,7 +935,7 @@ def get(self):
 class UnitUsers(flask_restful.Resource):
     """List unit users."""
 
-    @auth.login_required(role=["Unit Admin", "Unit Personnel"])
+    @auth.login_required(role=["Super Admin", "Unit Admin", "Unit Personnel"])
     @logging_bind_request
     def get(self):
         """Get and return unit users within the unit the current user is connected to."""
```
Comment (talavis) [inline dds_web/api/user.py:941]:
```suggestion
        """List unit users within the unit the current user is connected to, or the one defined by a superadmin."""
```

**Review by talavis — APPROVED** [review]

**Comment by inaod568** [comment]
Changed back from walrus operator because it requires Python 3.8. Definite improvement for the future but not limiting the dds to > 3.8 because of only this.

**Review by valyo — APPROVED** [review]

**Review by aanil — APPROVED** [review]
👍

**Pull request merged**